Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterOperationImpl.java

```diff
@@ -51,25 +50,23 @@
       */
      public List<ClusterRawInfo> listCluster(final SubscriptionDetail subscription) throws IOException, HDIException, AzureCmdException {
           try {
-               String response = requestWithToken(subscription.getTenantId(), new RequestCallback<String>() {
-                    @Override
-                    public String execute(String accessToken) throws Throwable {
-                         return AzureAADHelper.executeRequest(CommonConstant.HDINSIGHT_CLUSTER_URI,
-                                 String.format("api/Clusters/GetAll?subscriptionIds=%s;&_=%d", subscription.getSubscriptionId(), new Date().getTime()),
-                                 RestServiceManager.ContentType.Json,
-                                 "GET",
-                                 null,
-                                 accessToken,
-                                 new RestServiceManagerBaseImpl());
-                    }
-               });
+              String response = requestWithToken(subscription.getTenantId(), (accessToken) -> {
+                      String managementUrl = AuthMethodManager.getInstance().getAzureManager().getManagementURI();
```

Comment:
Instead of getManagementURi like this would suggest to use AuthMethodManager.getInstance().getAzureManager().getEnvironment() first to get the azure environment.

Reply:
Fixed. Thanks.